src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -555,17 +588,58 @@ private void transferAlignmentInfoToFragment(final SAMRecord unaligned, final SA
             log.warn("Record mapped off end of reference; making unmapped: " + aligned);
             SAMUtils.makeReadUnmapped(unaligned);
         } else if (isContaminant) {
-            log.warn("Record looks like foreign contamination; making unmapped: " + aligned);
-            // NB: for reads that look like contamination, just set unmapped flag and zero MQ but keep other flags as-is.
-            // this maintains the sort order so that downstream analyses can use them for calculating evidence
-            // of contamination vs other causes (e.g. structural variants)
+            if (numCrossSpeciesContaminantWarnings < MAX_CROSS_SPECIES_CONTAM_WARN) {
+                log.warn("Record looks like cross-species contamination ; unmapping read: " + aligned);
+                numCrossSpeciesContaminantWarnings++;
+            } else if (numCrossSpeciesContaminantWarnings == MAX_CROSS_SPECIES_CONTAM_WARN) {
+                log.warn("No more warnings regarding cross-species contamination; limit reached.");
+            }
+
+            // For reads that look like contamination, set unmap, and copy the mapping information to
+            // a tag since CRAM seems to throw away the mapping information for
+            // unaligned reads.
```

Comment:
Move to method documentation?  Perhaps also include in the description of the enum.